src/util/metrics/threads_linux.rs

```diff
@@ -15,42 +15,50 @@ use std::fs;
 use std::io::{Error, ErrorKind, Read, Result};
 use std::sync::Mutex;
 
-use prometheus::core::{Collector, Desc};
-use prometheus::{self, proto, CounterVec, Opts};
-
 use libc::{self, pid_t};
+use prometheus::core::{Collector, Desc};
+use prometheus::{self, proto, CounterVec, IntGaugeVec, Opts};
 
 /// monitor current process's threads.
 pub fn monitor_threads<S: Into<String>>(namespace: S) -> Result<()> {
     let pid = unsafe { libc::getpid() };
     let tc = ThreadsCollector::new(pid, namespace);
-    prometheus::register(Box::new(tc)).map_err(|e| to_err(format!("{:?}", e)))?;
-
-    Ok(())
+    prometheus::register(Box::new(tc)).map_err(|e| to_io_err(format!("{:?}", e)))
 }
 
+/// A collector to collect threads metrics, including CPU usage
+/// and threads state.
 struct ThreadsCollector {
-    pid: pid_t,
+    pid: Mutex<pid_t>,
```

Comment:
Why do we need `Mutex` here?

Reply:
I don't get it, what do you synchronize?

Reply:
The `collect` method, to be precise, the CPU time, https://github.com/pingcap/tikv/pull/3173/files#diff-21ec373afb597cc6661bb2a2eb148e52R85

Reply:
I am confused that why using Mutex for pid? seem it won't be changed.

Reply:
Yes, it wont be changed. It's a lock for synchronous the `collect`. Of course, we can also choose other fileds.